### ISSUE TITLE
Utilize Diffy for Visual Regression Testing

### ISSUE THREAD

**Issue by quicksketch** [backdrop-ops/backdropcms.org#1025]
The owner of [Diffy](https://diffy.website/), a visual regression testing service, has offered their services pro-bono to the Backdrop project. @stpaultim, @izmeez, and myself (@quicksketch) met with Yuriy from Diffy today to discuss it.

This issue is to coordinate the evaluation and setup process.

**Comment by ygerasimov** [comment]
As agreed on the call we will set up Diffy for the main website https://backdropcms.org/. It was great talking to you all today.

**Comment by stpaultim** [comment]
We have an account set up and have configured it to take snapshots of beta.backdropcms.org and backdropcms.org and compare them.

This is a pretty basic implementation of Diffy, but a good place to begin.

We'll talk more about this at our user group meeting, next Thursday.

**Comment by ygerasimov** [comment]
@stpaultim I see that staging environment miss images. Can we set up sync of the images from prod or maybe even better to have stage_file_proxy. Otherwise Diffy will show a lot of changes because of the missing images.

**Comment by stpaultim** [comment]
@ygerasimov - Yes, we'll look at that ASAP. I noticed this as well and am aware that the two sites are quite out of sync right now. So, we need to refresh the database on Beta.

Baby steps...

:-)

Atten: @bugfolder

**Comment by bugfolder** [comment]
> So, we need to refresh the database on Beta.

It's a little more than just that, I think. Since the dbs of Beta (site) depend on the code in beta (the branch) and the files on beta (the server), the following all should be done (preferably at one time):

* Update _beta_ branch of code from _main_ branch
* Backup and copy files from b.org to beta.b.org
* Copy config files from live-active on b.org to beta-active on beta.b.org
* Backup the Backdrop db on b.org, import to Backdrop db on beta.b.org
* Backup the CiviCRM db on b.org, import to CiviCRM db on beta.b.org
* Fix any permissions that didn't end up with the right owner or group

Currently, b.org has a bunch of available updates: Backdrop 1.26.2, which just came out, of course, but also 9 contrib modules, plus CiviCRM that also need to be updated to the latest versions. It's not strictly necessary to update those before mirroring from b.org, but if the goal is to have Beta as close as possible to b.org (at least for a little while; the divergence starts almost immediately since b.org gets several spam account creations per day), it might be desirable to do all the b.org updates first, then carry out the mirroring process.

Here's the release checklist for 1.26.2: https://github.com/backdrop/backdrop-issues/issues/6315

**Comment by bugfolder** [comment]
> Does the database need to be sanitized before importing into beta site?

No, beta site contains real user data (hence the hiding of it behind a password).

**Comment by klonos** [comment]
Hey @ygerasimov 👋🏼 ...nice to see you here (do you remember me from Salsa Digital?). Thank you for offering your services pro-bono to our project 🙏🏼 

@bugfolder are we using https://github.com/backdrop-contrib/stage_file_proxy on the beta site? We should! It would reduce the tasks that we need to be manually doing.

**Comment by klonos** [comment]
...also, can I please be added as a member to the team/organization in the Backdrop Diffy account?

**Comment by ygerasimov** [comment]
@klonos 👋🏼 great to see you here! Please advise your email address (drop me an email to info@diffy.website) and I'll send you the invite.
